### PR TITLE
geo/geogfn: implement ST_Azimuth({geometry,geometry})

### LINKED ISSUE #48887: geo/geomfn: implement ST_Azimuth({geometry,geometry})
Implement `ST_Azimuth` on arguments {geometry,geometry}, which should adopt [PostGIS behaviour](https://postgis.net/docs/ST_Azimuth.html).

_Observers: Please react to this issue if you need this functionality._

For Geometry builtins, please do the following:
* Ideally add a relevant helper function in [`pkg/geo/geomfn`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo/geomfn) (parse and output related functions can go in [`pkg/geo`](https://github.com/cockroachdb/cockroach/tree/master/pkg/geo)). Add exhaustive unit tests here - you can run through example test cases and make sure that PostGIS and CRDB return the same result within a degree of accuracy (1cm for geography).
  * When using GEOS, you can reference the [C API](https://github.com/libgeos/geos/blob/master/capi/geos_c.h.in) for which functions are available. Unfortunately, Windows is not currently supported when using GEOS.
* Create a new builtin that references this function in [`pkg/sql/sem/builtins/geo_builtins.go`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/sem/builtins/geo_builtins.go). Note that we currently do not support optional arguments, so we define functions that have optional arguments once without the optional argument (using the default value in the optional position), and once with the optional argument.
* Modify the tests in [`pkg/sql/logictest/testdata/logic_test/geospatial`](https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/logictest/testdata/logic_test/geospatial) to call this functionality at least once. You can call `make testbaselogic FILES='geospatial' TESTFLAGS='-rewrite'` to regenerate the output. Tests here should just ensure the builtin is linked end to end.
* Ensure the documentation is regenerated by calling `make buildshort`.
* Submit your PR - make sure to follow the guidelines from [creating your first PR](https://wiki.crdb.io/wiki/spaces/CRDB/pages/181633464/Your+first+CockroachDB+PR]).

You can follow #48552 for an example PR.

<sub>:robot: This issue was synced with a spreadsheet by [gsheets-to-github-issues](https://github.com/cockroachlabs/gsheet-to-github-issues) by [otan](https://github.com/otan) on 2020-06-02T05:16:25Z. Changes to titles, body and labels may be overwritten.</sub>

### PR DESCRIPTION
**Pull request by hueypark**
Fixes #48887

Release note (sql change): This PR implement adds the ST_Azimuth({geometry,geometry})

### CODE REVIEW

File: pkg/geo/geomfn/azimuth_test.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+	"github.com/stretchr/testify/require"
+	"github.com/twpayne/go-geom"
+)
+
+var (
+	zero            = 0.0
+	aQuarterPi      = 0.7853981633974483
+	towQuartersPi   = 1.5707963267948966
+	threeQuartersPi = 2.356194490192344
```
Comment (otan) [inline pkg/geo/geomfn/azimuth_test.go:25]: do you mind moving these vars into the `TestAzimuth` function? (unless we're planning to re-use them)

File: pkg/geo/geomfn/azimuth.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+// Azimuth returns the azimuth in radians of the segment defined by the given point geometries.
+// The azimuth is angle is referenced from north, and is positive clockwise.
+// North = 0; East = π/2; South = π; West = 3π/2.
+func Azimuth(a *geom.Point, b *geom.Point) *float64 {
+	if a.X() == b.X() && a.Y() == b.Y() {
+		return nil
+	}
+
+	azimuth := math.Mod(2*math.Pi+math.Pi/2-math.Atan2(b.Y()-a.Y(), b.X()-a.X()), 2*math.Pi)
```
Comment (otan) [inline pkg/geo/geomfn/azimuth.go:27]: do you mind adding a comment of why we're adding 2*math.Pi, i.e. we always want a positive number?
some commentary on that math would also be nice :)

**Review by otan — COMMENTED** [review]
quick comment request :)

File: pkg/geo/geomfn/azimuth.go [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"math"
+
+	"github.com/twpayne/go-geom"
+)
+
+// Azimuth returns the azimuth in radians of the segment defined by the given point geometries.
+// The azimuth is angle is referenced from north, and is positive clockwise.
+// North = 0; East = π/2; South = π; West = 3π/2.
+func Azimuth(a *geom.Point, b *geom.Point) *float64 {
```
Comment (otan) [inline pkg/geo/geomfn/azimuth.go:22]: actually, do you mind moving the logic in geo_builtins such that Azimuth only takes in `(a *geo.Geometry, b *geo.Geometry)`, to be consistent with the rest of the function signatures in ehre?

**Comment by hueypark** [comment]
Thank you for the review!

File: pkg/geo/geomfn/azimuth_test.go (hunk 1 above)
Comment (hueypark) [reply pkg/geo/geomfn/azimuth_test.go]: Done.

File: pkg/geo/geomfn/azimuth.go (hunk 2 above)
Comment (hueypark) [reply pkg/geo/geomfn/azimuth.go]: Done.

File: pkg/geo/geomfn/azimuth.go (hunk 3 above)
Comment (hueypark) [reply pkg/geo/geomfn/azimuth.go]: Done.

**Review by otan — APPROVED** [review]
thanks for your contribution!

**Pull request merged**